pkg/log/logshipper/logshipper.go

```diff
@@ -0,0 +1,77 @@
+package logshipper
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/kolide/launcher/pkg/agent/types"
+	"github.com/kolide/launcher/pkg/sendbuffer"
+)
+
+// TODO: consolidate this var, but where?
+var observabilityIngestTokenKey = []byte("observability_ingest_auth_token")
+
+type LogShipper struct {
+	sender     *authedHttpSender
+	sendBuffer *sendbuffer.SendBuffer
+	logger     *logger
+	knapsack   types.Knapsack
+	ctx        context.Context
+	cancel     context.CancelFunc
+}
+
+func New(ctx context.Context, k types.Knapsack) *LogShipper {
```

Comment:
Why does new take a ctx?

Reply:
due to the signature required for rungroup.Execute, wish we could pass ctx there ... I could just do an inline func for execute in the launcher start up, I'll try that and see how it feels

Reply:
ehh.. not sure about this one. We could just create a new context with background as the parent, but would like to also stop if the parent context gets canceled 🤷

Reply:
what do you think?

Reply:
I usually expect `New` to setup things that cannot be canceled, and `Run` to take a cancelable context.

Eg: new just creates a struct, nothing runtimey.